### PR TITLE
Implementing MVVM architecture in android using Kotlin

### LINKED ISSUE #559: Implementing MVVM Architecture in Android using kotlin
# Brief Summary: 
MVVM or Model-View-Viewmodel is an architectural pattern that allows developers to separate their business logic from the user interface. It makes it easy to perform CRUD (create, read, update, and delete) functions. Adopting MVVM architecture can help in the creation of high quality apps. The article provides a detailed example to help you understand MVVM architecture.

# Key Takeaways: 

Understand the MVVM architecture
Incorporate MVVM architecture in their future apps
Debug errors resulting from the MVVM architecture

# References: 
https://developer.android.com/jetpack/guide

### PR DESCRIPTION
**Pull request by WanjaMIKE**
This fixes #559 
Implementing MVVM architecture in android using Kotlin

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tool:** [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by kayere** [comment]
Hello @WanjaMIKE, this a great topic. I will be reviewing your article. The review process involves two phases. I will be checking at grammar and readability in the first phase. The second phase involves checking for technical accuracy. As I go through the first review, please make sure to add your author details.

**Comment by WanjaMIKE** [comment]
okay, thank you

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by WanjaMIKE** [comment]
Okay

File: articles/implementing-mvvm-architecture-in-android-using-kotlin/index.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#### The goal of the tutorial
+By the end of this tutorial, you will create an app that takes input and displays it on a recycler view. A screenshot of the app is shown below.
+
+### Step 1 – Launching Android Studio.
+Launch Android Studio and create a new project, as shown below. Make sure that you select Kotlin as your preferred programming language. If you do not have Android Studio, you can install it from [here](https://developer.android.com/studio?gclid=Cj0KCQiA7qP9BRCLARIsABDaZzieBJWjBnokDdH6b0gQchoqudRXNohAGp_noSqALLuSlYuwA6EB5T4aAntwEALw_wcB&gclsrc=aw.ds)
+
+![getting started](/engineering-education/implementing-mvvm-architecture-in-android-using-kotlin/getting-started.png)
+
+### Step 2 – Creating the model.
+Create the app model. This is also referred to as the data class. To avoid confusion, create a package named model inside the java folder. Then, create a data class named Blog in the model package as shown below. For simplicity, the data class will only have one variable (title). In Kotlin, there is no need to use the get and set keywords. These functionalities are added to the data class automatically.
```
Comment (kayere) [inline articles/implementing-mvvm-architecture-in-android-using-kotlin/index.md:24]: Rephrase the last sentence to this "There is no need to add getters and setters, Kotlin adds them to the class automatically." It is shorter and also eliminates the use of passive voice.

File: articles/implementing-mvvm-architecture-in-android-using-kotlin/index.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+![model](/engineering-education/implementing-mvvm-architecture-in-android-using-kotlin/model.png)
+
+### Step 3 – Creating the view.
+The view is what the user sees on the screen. The UI, therefore, needs to be well structured to minimize confusion and dissatisfaction.
+
+Open `activity_main.xml` file and change the layout from constraint to linear layout. Ensure that the orientation is set to vertical. This ensures that the UI components such as buttons and text views are arranged vertically. For our app, the primary widgets are `Edittext`, `Button`, and a `RecyclerView`. Ensure that all these widgets have IDs since they will be required at a later stage. The layout design and code are shown in the image below. The section with a list showing items 0 to 9 is the `RecyclerView`. Whatever we type in the app will be displayed in that section once the submit button is clicked.
```
Comment (kayere) [inline articles/implementing-mvvm-architecture-in-android-using-kotlin/index.md:31]: Replace "This ensures that the UI components... " with "This arranges UI components vertically in the Layout.". It easier to read.

File: articles/implementing-mvvm-architecture-in-android-using-kotlin/index.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+![view](/engineering-education/implementing-mvvm-architecture-in-android-using-kotlin/view.png)
+
+### Step 4 – Creating the view.
+Still, on the layout, we need to create the design of the element that will be shown in the `RecyclerView`. Therefore, create a file named `item.xml` and add the code shown in the image below. The design is simple since only one attribute from the data class will be displayed to the user.
+
+![itemview](/engineering-education/implementing-mvvm-architecture-in-android-using-kotlin/itemview.png)
+
+### Step 5 – Create a RecyclerView Adapter
+A `RecyclerView` adapter is responsible for binding the `item.xml` layout to the `RecyclerView`. It also takes in a list of items and displays them to the user. The code for the `RecyclerView` adapter is shown below.
```
Comment (kayere) [inline articles/implementing-mvvm-architecture-in-android-using-kotlin/index.md:41]: Replace " is responsible for" with "handles"

**Review by kayere — CHANGES_REQUESTED** [review]
@WanjaMIKE - I have done a preliminary review on your article. Grammar and readability is fair. I have made some syntax and readability changes. I have also suggested some changes for you to make. The article has a lot of passive voice use and some few hard to read sentences. I have fixed some of them for you. Use [Hemingway](http://www.hemingwayapp.com/) to identify and fix the rest. Make the changes so that we can proceed to the tech review. Don't forget to pull the changes I have made first.

**Comment by WanjaMIKE** [comment]
Okay, thank you

**Comment by WanjaMIKE** [comment]
Changes fixed

**Comment by kayere** [comment]
> Changes fixed

I can't see the changes. Please make sure you push your commit.

**Comment by WanjaMIKE** [comment]
I have pushed the file. thanks

**Review by kayere — COMMENTED** [review]
@WanjaMIKE - I have gone through the tech review, and I have some suggestions. Please use code snippets rather than images. This allows readers to copy your code. The other suggestion is on the inline comment. Make the changes and the article will be good to go.

**Comment by kayere** [comment]
*Note* - Don't forget to add the Author details on your next commit.

**Comment by WanjaMIKE** [comment]
Okay I have made changes

**Comment by kayere** [comment]
> Okay I have made changes

Push your commit.

**Comment by WanjaMIKE** [comment]
> > Okay I have made changes
> 
> Push your commit.

I have

**Review by kayere — COMMENTED** [review]
@WanjaMike - I am almost done with the Tech review, but I have one last suggestion. Please explain what the functions in the `MainActivity` class do.

**Review by kayere — APPROVED** [review]
@hectorkambow - I am done with the Tech review for this article. Please give it a final review.

**Comment by hectorkambow** [comment]
@WanjaMIKE  great 1st article - it looks like it will be a helpful addition to our EngEd community - please go over any of the feedback (or changes) made by the peer reviewers and/or myself to see how we can apply that same level of quality to all future articles 🚀 
Article will be published momentarily  👍

**Comment by hectorkambow** [comment]
@kayere  👍  Thank you for the great review 🚀  keep up the great work.

**Comment by WanjaMIKE** [comment]
Okay, thank you

**Pull request merged**